src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs

```diff
@@ -41,10 +41,11 @@ private class ParsedSyntaxTree : CSharpSyntaxTree
             {
                 Debug.Assert(root != null);
                 Debug.Assert(options != null);
-                Debug.Assert(textOpt == null || textOpt.Encoding == encodingOpt && textOpt.ChecksumAlgorithm == checksumAlgorithm);
+                Debug.Assert(text == null || text.Encoding == encoding && text.ChecksumAlgorithm == checksumAlgorithm);
+                Debug.Assert(SourceHashAlgorithms.IsSupportedAlgorithm(checksumAlgorithm));
```

Comment:
Is this the only real change in this file? Consider reverting the naming changes.

Reply:
This and overriding the ChecksumAlgorithm